scripts/exglobal_prep_ocean_obs.py

```diff
@@ -24,30 +25,51 @@
 OCNOBS2IODAEXEC = os.getenv('OCNOBS2IODAEXEC')
 
 OBS_YAML = os.getenv('OBS_YAML')
+# this will fail with FileNotFoundError if all yaml files in OBS_YAML are not
+# present in OBS_YAML_DIR
 obsConfig = YAMLFile(OBS_YAML)
 
 OBSPROC_YAML = os.getenv('OBSPROC_YAML')
 obsprocConfig = YAMLFile(OBSPROC_YAML)
 
 # TODO (AFE): needs more error handling (missing sources, missing files)
-for observer in obsConfig['observers']:
+try:
+    # For each of the observation sources (observers) specificed in the OBS_YAML...
+    for observer in obsConfig['observers']:
 
-    obsSpaceName = observer['obs space']['name']
-    print(f"obsSpaceName: {obsSpaceName}")
+        try:
+            obsSpaceName = observer['obs space']['name']
+        except KeyError:
+            print(f"observer: {observer}")
+            print("WARNING: Ill-formed observer yaml file, skipping")
+            continue  # to next observer
 
-    for observation in obsprocConfig['observations']:
+        print(f"obsSpaceName: {obsSpaceName}")
```

Comment:
Shouldn't this be inside the try? The exception won't define this variable and then it will fail...

Reply:
This gets the `observer` from `OBS_YAML`, which looks like this:
```
observers:
- !INC ${OBS_YAML_DIR}/adt_all.yaml
- !INC ${OBS_YAML_DIR}/adt_j3.yaml
```
This `try` fails when the individual observer yaml file can't provide `observer['obs space']['name']`, and then moves on to the next one